site/documentation/static/api/device-registry-v1.yaml

```diff
@@ -527,23 +527,25 @@ components:
 
       CredentialsSet:
          type: array
+         description: A set of credentials. The entries in this list must be
+                      unique by the composite key of `auth-id` and `type`.
          items:
-            $ref: '#/components/schemas/TypedSecret'
+            $ref: '#/components/schemas/TypedCredentials'
 
-      TypedSecret:
+      TypedCredentials:
          additionalProperties: false
          oneOf:
-            - $ref: '#/components/schemas/PasswordSecret'
-            - $ref: '#/components/schemas/PSKSecret'
-            - $ref: '#/components/schemas/X509CertificateSecret'
+            - $ref: '#/components/schemas/PasswordCredentials'
+            - $ref: '#/components/schemas/PSKCredentials'
+            - $ref: '#/components/schemas/X509CertificateCredentials'
```

Comment:
This looks like it would restrict the possible content to just these three types, wouldn't it? The original idea was to allow for any kind of secrets to be supported. Can this be defined in a way that allows additional types to be used as well?

Reply:
I'm not sure we could have a useful "type" for some unknown future enhancement. I think the spec should evolve with the new requirements when we know them

Reply:
Of course, once we know about a new type, we can add it, sure. I was just wondering about what to do in the mean time if somebody implementing the API wants to use a (custom?) credentials type. If this spec is just meant for (informal) reference then I think we do not have a problem. However, if some payload validation is supposed to be based on the JSON schema then I think we should provide for extensibility and/or toleration of unexpected content. What's your take?

Reply:
I'm not sure that we can have proper validation without a defined type and how that would help someone implement a registry without defined credential type. I of course, could be wrong. Can you explain how would you define this?